src/messages/proposals.rs

```diff
@@ -233,17 +236,33 @@ impl<'a> ProposalQueue<'a> {
     pub(crate) fn from_committed_proposals(
         ciphersuite: &Ciphersuite,
         committed_proposals: &'a [ProposalOrRef],
-        proposals_by_reference: &'a ProposalQueue<'a>,
+        proposals_by_reference: &[&'a MLSPlaintext],
         sender: Sender,
     ) -> Result<Self, ProposalQueueError> {
+        // Feed the `proposals_by_reference` in a `HashMap` so that we can easily
+        // extract then by reference later
+        let mut proposals_by_reference_queue: HashMap<ProposalReference, QueuedProposal> =
+            HashMap::new();
```

Comment:
You can put the `<ProposalReference, QueuedProposal>` here to the new and save yourself a `HashMap`.

Reply:
Sorry, I'm not sure what you're suggesting here.

Reply:
the following, equivalent, is a little shorter
`let mut proposals_by_reference_queue = HashMap::<ProposalReference, QueuedProposal>::new();`